File: sdks/php/src/Types/WalletTransactionResponse.php
<?php

namespace NovaBilling\Types;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class WalletTransactionResponse extends JsonSerializableType
{
    /**
     * @var string $id
     */
    #[JsonProperty('id')]
    public string $id;

    /**
     * @var string $walletId
     */
    #[JsonProperty('walletId')]
    public string $walletId;

    /**
     * @var value-of<WalletTransactionResponseTransactionType> $transactionType
     */
    #[JsonProperty('transactionType')]
    public string $transactionType;

    /**
     * @var value-of<WalletTransactionResponseStatus> $status
     */
    #[JsonProperty('status')]
    public string $status;

    /**
     * @var value-of<WalletTransactionResponseTransactionStatus> $transactionStatus
     */
    #[JsonProperty('transactionStatus')]
    public string $transactionStatus;

    /**
     * @var string $creditAmount Credits added or deducted
     */
    #[JsonProperty('creditAmount')]
    public string $creditAmount;

    /**
     * @var string $amount Monetary equivalent
     */
    #[JsonProperty('amount')]
    public string $amount;

    /**
     * @var ?string $invoiceId
     */
    #[JsonProperty('invoiceId')]
    public ?string $invoiceId;

    /**
     * @var ?string $settledAt
     */
    #[JsonProperty('settledAt')]
    public ?string $settledAt;

    /**
     * @var ?array<string, mixed> $metadata
     */
    #[JsonProperty('metadata'), ArrayType(['string' => 'mixed'])]
    public ?array $metadata;

    /**
     * @var string $createdAt
     */
    #[JsonProperty('createdAt')]
    public string $createdAt;

    /**
     * @param array{
     *   id: string,
     *   walletId: string,
     *   transactionType: value-of<WalletTransactionResponseTransactionType>,
     *   status: value-of<WalletTransactionResponseStatus>,
     *   transactionStatus: value-of<WalletTransactionResponseTransactionStatus>,
     *   creditAmount: string,
     *   amount: string,
     *   createdAt: string,
     *   invoiceId?: ?string,
     *   settledAt?: ?string,
     *   metadata?: ?array<string, mixed>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->id = $values['id'];
        $this->walletId = $values['walletId'];
        $this->transactionType = $values['transactionType'];
        $this->status = $values['status'];
        $this->transactionStatus = $values['transactionStatus'];
        $this->creditAmount = $values['creditAmount'];
        $this->amount = $values['amount'];
        $this->invoiceId = $values['invoiceId'] ?? null;
        $this->settledAt = $values['settledAt'] ?? null;
        $this->metadata = $values['metadata'] ?? null;
        $this->createdAt = $values['createdAt'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/rust/src/api/types/wallet_transaction_response.rs
pub use crate::prelude::*;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct WalletTransactionResponse {
    pub id: String,
    #[serde(rename = "walletId")]
    pub wallet_id: String,
    #[serde(rename = "transactionType")]
    pub transaction_type: WalletTransactionResponseTransactionType,
    pub status: WalletTransactionResponseStatus,
    #[serde(rename = "transactionStatus")]
    pub transaction_status: WalletTransactionResponseTransactionStatus,
    /// Credits added or deducted
    #[serde(rename = "creditAmount")]
    pub credit_amount: String,
    /// Monetary equivalent
    pub amount: String,
    #[serde(rename = "invoiceId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub invoice_id: Option<String>,
    #[serde(rename = "settledAt")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub settled_at: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<HashMap<String, serde_json::Value>>,
    #[serde(rename = "createdAt")]
    pub created_at: String,
}